src/gui/editors/PianoRoll.cpp

```diff
@@ -712,6 +712,85 @@ void PianoRoll::glueNotes()
 	}
 }
 
+void PianoRoll::fitNoteLengths(bool fill)
+{
+	if (!hasValidPattern()) { return; }
+	m_pattern->addJournalCheckPoint();
+
+	// Reference notes
+	NoteVector refNotes = m_pattern->notes();
+	std::sort(refNotes.begin(), refNotes.end(), Note::lessThan);
+
+	// Notes to edit
+	NoteVector notes = getSelectedNotes();
+	if (notes.empty())
+	{
+		notes = refNotes;
+	}
+	else if (!fill)
+	{
+		std::sort(notes.begin(), notes.end(), Note::lessThan);
+	}
+	if (fill)
+	{
+		std::sort(notes.begin(), notes.end(), [](Note *n1, Note *n2) { return n1->endPos() < n2->endPos(); });
+	}
+
+	int length;
+	NoteVector::iterator ref = refNotes.begin();
+	for (Note *note : notes)
```

Comment:
```suggestion
	for (Note* note : notes)
```

Reply:
Is this LMMS' preferred style? Then it should be added to the [Coding Conventions](https://github.com/LMMS/lmms/wiki/Coding-conventions). It currently says if a convention is not mentioned there it may fall under the Qt coding style, which in this case is asterisk to the right.

Reply:
I think it wasn't added yet, only discussed on Discord. Maybe I should bring it up about adding it to the wiki page as well.